video_xblock/mixins.py

```diff
@@ -165,6 +195,81 @@ def convert_3playmedia_caps_to_vtt(self, caps, video_id, lang="en", lang_label="
             response = {"lang": lang, "url": external_url, "label": lang_label}
         return response
 
+    def fetch_available_3pm_transcripts(self):
+        """
+        Fetch all available transcripts from 3PlayMedia API for current file ID.
+
+        :return: (generator of OrderedDicts) all transcript's data
+        """
+        results = self.get_available_3pm_transcripts(self.threeplaymedia_file_id, self.threeplaymedia_apikey)
+        errors = isinstance(results, dict) and results.get('errors')
+        log.debug("Fetched 3PM transcripts:\n{}".format(results))
+        if errors:
+            log.error("3PlayMedia transcripts fetching API request has failed!\n{}".format(
+                results.get('errors')
+            ))
+            raise StopIteration
+        for transcript_data in results:
+            transcript = self.fetch_3pm_translation(transcript_data)
+            transcript_ordered_dict = transcript._asdict()
```

Comment:
1) Please add test for empty transcript
2) Think if we need to show this error to user.